lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint.py

```diff
@@ -0,0 +1,384 @@
+    if not module.params.get('vpc_id'):
+        module.fail_json(msg='vpc_id is a required paramater')
+    else:
+        vpc_id = module.params.get('vpc_id')
+    if not module.params.get('service'):
+        module.fail_json(msg='a valid service is a required paramater')
+    else:
+        service_name = module.params.get('service')
+
+    if module.params.get('route_table_ids'):
+        route_table_ids = module.params.get('route_table_ids')
+        existing_endpoints = get_endpoints(client, module)
+        for endpoint in existing_endpoints['VpcEndpoints']:
+            if endpoint['VpcId'] == vpc_id and endpoint['ServiceName'] == service_name:
+                sorted_endpoint_rt_ids = sorted(endpoint['RouteTableIds'])
+                sorted_route_table_ids = sorted(route_table_ids)
+                if cmp(sorted_endpoint_rt_ids, sorted_route_table_ids) == 0:
+                    return False, camel_dict_to_snake_dict(endpoint)
+
+    changed, result = create_vpc_endpoint(client, module)
+
+    return changed, json.loads(json.dumps(result, default=date_handler))
+
+
+def create_vpc_endpoint(client, module):
+    params = dict()
+    changed = False
+    token_provided = False
+    params['VpcId'] = module.params.get('vpc_id')
+    params['ServiceName'] = module.params.get('service')
+    params['DryRun'] = module.check_mode
+
+    if module.params.get('route_table_ids'):
+        params['RouteTableIds'] = module.params.get('route_table_ids')
+
+    if module.params.get('client_token'):
+        token_provided = True
+        request_time = datetime.datetime.utcnow()
+        params['ClientToken'] = module.params.get('client_token')
+
+    policy = None
+    if module.params.get('policy'):
+        try:
+            policy = json.loads(module.params.get('policy'))
+        except ValueError as e:
+            module.fail_json(msg=str(e), exception=traceback.format_exc())
+
+    elif module.params.get('policy_file'):
+        try:
+            with open(module.params.get('policy'), 'r') as json_data:
+                policy = json.load(json_data)
+        except Exception as e:
+            module.fail_json(msg=str(e), exception=traceback.format_exc())
+
+    if policy:
+        params['PolicyDocument'] = json.dumps(policy)
+
+    try:
+        changed = True
+        result = camel_dict_to_snake_dict(client.create_vpc_endpoint(**params)['VpcEndpoint'])
+        if token_provided and (request_time > result['creation_timestamp'].replace(tzinfo=None)):
+            changed = False
+        elif module.params.get('wait') and not module.check_mode:
+            status_achieved, result = wait_for_status(client, module, result['vpc_endpoint_id'], 'available')
+            if not status_achieved:
+                module.fail_json(msg='Error waiting for vpc endpoint to become available - please check the AWS console')
+    except botocore.exceptions.ClientError as e:
+        if "DryRunOperation" in e.message:
```

Comment:
For all below - Handle exception as per guidelines https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2